node/building_blocks.re

```diff
@@ -96,18 +119,62 @@ let is_signable = (state, block) => {
   && is_current_producer(state, ~key_hash=block.author)
   && !has_next_block_to_apply(state, ~hash=block.hash)
   && all_main_ops_are_known
-  && contains_only_trusted_add_validator_op(block.side_chain_ops);
+  && contains_only_trusted_add_validator_op(block.side_chain_ops)
+  && block_has_signable_state_root_hash(~current_time, state, block);
 };
 
 let sign = (~key, block) => Block.sign(~key, block);
 
-let produce_block = state =>
+/** The state_root_min_timeout determines how often the block producer
+    starts hashing a new state root. It is closely related to the
+    state root epoch (see notes in [should_start_new_epoch]]).
+
+    The faster hashing is, the shorter this time should be. Faster updates
+    to the state root hash means shorter times to join the network, since a
+    node only needs to download and apply the blocks from the previous state root
+    update to the current block.
```

Comment:
"since a node needs to wait a state root hash change on Tezos before being considered in sync"